src/core/net/ip6.cpp

```diff
@@ -1267,7 +1267,7 @@ Error Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkMes
                     break;
                 }
             }
-            VerifyOrExit(isAllowedType, error = kErrorDrop);
+            VerifyOrExit(isAllowedType, error = kErrorNone);
```

Comment:
The original code check if the error is `kErrorDrop`, and use NOTE instead of WARN for `kErrorDrop`. 
However, it's difficult to be certain that all `kErrorDrop` errors are not significant. 
So, it is better to return `kErrorNone` for cases that we are sure that it's fine?

Reply:
In this case, we are indeed dropping the message, so I think we should keep the error.

I know it is an easy way to skip logging in this case and probably works fine for now but changing the returned error and returning `kErrorNone` when this method did indeed drop the message can lead to hard-to-debug issues in future if/when the `HandleDatagram()` returned value is used when it is called. New code relying on returned `error` may not notice this change here.

Reply:
Don't have strong preference for either solution.

Reverted to the original code.